core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala

```diff
@@ -44,35 +44,34 @@ import RestApiCommons._
  * Abstract class which provides basic Directives which are used to construct route structures
  * which are common to all versions of the Rest API.
  */
-abstract protected[controller] class RestAPIVersion(
-    protected val apiversion: String,
-    protected val build: String,
-    protected val buildno: String)
+protected[controller] class SwaggerDocs(
+    apipath: String,
+    doc: String)(
+        implicit actorSystem: ActorSystem)
     extends Directives {
 
-    /** Base API prefix. */
-    protected val apipath = "api"
-
     /** Swagger end points. */
     protected val swaggeruipath = "docs"
     protected val swaggerdocpath = "api-docs"
 
-    def prefix = pathPrefix(apipath / apiversion)
-
     /**
-     * This is the most important method -- it provides the routes that define the REST API.
+     * Defines the routes to serve the swagger docs.
      */
-    def routes(implicit transid: TransactionId): Route
+    val swaggerRoutes: Route = {
+        pathPrefix(swaggeruipath) {
+            getFromDirectory("/swagger-ui/")
+        } ~ path(swaggeruipath) {
+            redirect(s"$swaggeruipath/index.html?url=$apiDocsUrl", PermanentRedirect)
+        } ~ pathPrefix(swaggerdocpath) {
+            pathEndOrSingleSlash {
+                getFromResource(doc)
+            }
+        }
+    }
 
-    /**
-     * Information which describes details of a particular deployment of the REST API.
-     */
-    def info = {
-        JsObject(
-            "openwhisk" -> "hello".toJson,
-            "version" -> apiversion.toJson,
-            "build" -> build.toJson,
-            "buildno" -> buildno.toJson)
+    private def apiDocsUrl = {
+        if (apipath.endsWith("/")) s"$apipath$swaggerdocpath"
+        else s"$apipath/$swaggerdocpath"
```

Comment:
Should we rather strongly type the `Path` vals to be `spray.http.Uri.Path` instead of `String`? Could prevent the "unsureness" here.